cms/djangoapps/contentstore/management/commands/tests/test_force_publish.py

```diff
@@ -0,0 +1,91 @@
+        """
+        Test 'force_publish' command with no arguments
+        """
+        errstring = "force_publish requires 1 argument: <course_id>"
+        with self.assertRaisesRegexp(CommandError, errstring):
+            self.command.handle()
+
+    def test_invalid_course_key(self):
+        """
+        Test 'force_publish' command with invalid course key
+        """
+        errstring = "Invalid course key."
+        with self.assertRaisesRegexp(CommandError, errstring):
+            self.command.handle('TestX/TS01')
+
+    def test_too_many_arguments(self):
+        """
+        Test 'force_publish' command with more than 1 arguments
+        """
+        errstring = "force_publish requires 1 argument: <course_id>"
+        with self.assertRaisesRegexp(CommandError, errstring):
+            self.command.handle(unicode(self.course.id), 'invalid-arg')
+
+    def test_course_key_not_found(self):
+        """
+        Test 'force_publish' command with non-existing course key
+        """
+        errstring = "Course not found."
+        with self.assertRaisesRegexp(CommandError, errstring):
+            self.command.handle(unicode('course-v1:org+course+run'))
+
+    def test_force_publish_non_split(self):
+        """
+        Test 'force_publish' command doesn't work on non split courses
+        """
+        course = CourseFactory(default_store=ModuleStoreEnum.Type.mongo)
+        errstring = 'The owning modulestore does not support this command.'
+        with self.assertRaisesRegexp(CommandError, errstring):
+            self.command.handle(unicode(course.id))  # pylint: disable=E1101
+
+    def test_force_publish(self):
+        """
+        Test 'force_publish' command
+        """
+        # Add some changes to course
+        chapter = ItemFactory.create(category='chapter', parent_location=self.course.location)
+        self.store.create_child(
+            self.user.id,
+            chapter.location,
+            'html',
+            block_id='html_component'
+        )
+
+        # verify that course has changes.
+        self.assertTrue(self.store.has_changes(self.store.get_item(self.course.location)))
+        # verify that draft and publish point to differnt versions
+        versions = get_course_versions(unicode(self.course.id))
+        self.assertNotEqual(versions['draft-branch'], versions['published-branch'])
+
+        with mock.patch('contentstore.management.commands.force_publish.query_yes_no') as patched_yes_no:
+            patched_yes_no.return_value = True
+            # force publish course
+            self.command.handle(unicode(self.course.id))
+
+            # verify that course has no changes
+            self.assertFalse(self.store.has_changes(self.store.get_item(self.course.location)))
+            # verify that draft and publish point to same versions
+            versions = get_course_versions(unicode(self.course.id))
+            self.assertEqual(versions['draft-branch'], versions['published-branch'])
```

Comment:
You also need to assert that the draft branch didn't change while the published branch did